src/SparkFun_u-blox_SARA-R5_Arduino_Library.h

```diff
@@ -852,22 +922,35 @@ class SARA_R5 : public Print
   SARA_R5_error_t sendHTTPPOSTfile(int profile, String path, String responseFilename, String requestFile, SARA_R5_http_content_types_t httpContentType);
 
   SARA_R5_error_t nvMQTT(SARA_R5_mqtt_nv_parameter_t parameter);
-  SARA_R5_error_t setMQTTclientId(String clientId);
-  SARA_R5_error_t setMQTTserver(String serverName, int port);
+  SARA_R5_error_t setMQTTclientId(const String& clientId);
```

Comment:
I think using String references rather than copied String instances is better for the health of the heap, unless the entire String object lives on the stack.

Aside from that I think taking a const String& both tells the caller the method won't change the String, and the method doesn't need to check if the argument is nullptr.

Reply:
I'm pretty sure this will be OK. I'll give it a quick test after merging.